syntax = "proto3";

package protocol.hotel;

import "protocol/types.proto";


message FetchRoomPayload {
  optional string startDate = 1;
  optional string endDate = 2;
  optional int32 beds = 3;
  optional int32 minsize = 4;
  optional int32 minprize = 5;
  optional int32 maxprice = 6;

}

message FetchRoomResponse {
  protocol.Header  header  = 1;
  repeated protocol.Room rooms = 2;
 

}

message protocolRequest {
  string uuid = 1;
  string start_date = 2;
  string end_date = 3;
}

message protocolResponse {
  protocol.Header  header  = 1;
  bool success = 2;
  string confirmation_code = 3;

}


service HotelService {

  rpc FetchRooms (FetchRoomPayload) returns (FetchRoomResponse);

  rpc Makeprotocol (protocolRequest) returns (protocolResponse);
}
